src/drag-drop-context.tsx

```diff
@@ -210,56 +342,59 @@ export const DragDropContext = (props) => {
       }
     }
   };
-
-  const displace = ({ type, id, transform }) => {
+  const displace = (
+    type: "draggables" | "droppables",
+    id: string,
+    transform: Transform
+  ): void => {
     untrack(() => {
       if (state[type][id]) {
-        setState(type, id, "transform", { ...transform });
+        setState(type, id, (value) => {
+          return value ? { ...value, transform: { ...transform } } : value;
+        });
       }
     });
   };
-
-  const dragStart = ({ draggableId }) => {
+  const dragStart = (draggableId: string): void => {
     batch(() => {
-      setState("draggables", draggableId, "transform", noopTransform());
+      setState("draggables", draggableId, (value) => {
+        return value ? { ...value, transform: noopTransform() } : value;
+      });
       setState("active", "draggable", draggableId);
     });
     recomputeLayouts();
     detectCollisions();
   };
-
-  const dragMove = ({ transform }) => {
+  const dragMove = (transform: Transform): void => {
     const draggableId = state.active.draggable;
     if (draggableId) {
-      setState("draggables", draggableId, "transform", { ...transform });
+      setState("draggables", draggableId, (value) => {
+        return value ? { ...value, transform: { ...transform } } : value;
+      });
       detectCollisions();
     }
   };
-
-  const dragEnd = () => {
+  const dragEnd = (): void => {
     batch(() => {
       setState("previous", "draggable", state.active.draggable);
       setState("previous", "droppable", state.active.droppable);
       if (state.active.draggable) {
-        setState("draggables", state.active.draggable, "transform", {
-          x: 0,
-          y: 0,
+        setState("draggables", state.active.draggable, (value) => {
+          return value ? { ...value, transform: noopTransform() } : value;
         });
       }
       setState("active", ["draggable", "droppable"], null);
     });
   };
-
-  const onDragStart = (handler) => {
+  const onDragStart = (handler: DragEventHandler): void => {
```

Comment:
**Major:** When passing custom data (via `createSortable`) the handler ends up with `draggable.data` being an empty proxy state. I checked against the latest published version (0.3.3) and it didn't happen there. 

Could the typings somehow cause this behaviour?

```js
const SortableItem = (props) => {
    const sortable = createSortable({ id: props.item.id, data: props.item });
    ...
} 

const [, { onDragStart }] = useDragDropContext();
onDragStart(({ draggable }) => {
    console.log(draggable.data) // Proxy { <target>: {}, <handler>: {…} }
}
```

Reply:
Nevermind - it is because of the signature change of `createSortable`. I was still passing in an object, which was interpreted as the `id` positional parameter. 

But now I'm curious - why is there no type warning for me here? Shouldn't my editor (vscode) be telling me I am passing in an `object` when a `string` is expected for `id`?